library/ssl_tls13_generic.c

```diff
@@ -368,6 +373,18 @@ static int ssl_tls13_parse_certificate_verify( mbedtls_ssl_context *ssl,
             md_alg = MBEDTLS_MD_SHA512;
             sig_alg = MBEDTLS_PK_ECDSA;
             break;
+#if defined(MBEDTLS_X509_RSASSA_PSS_SUPPORT)
+        case MBEDTLS_TLS13_SIG_RSA_PSS_RSAE_SHA256:
+            MBEDTLS_SSL_DEBUG_MSG( 4, ( "Certificate Verify: using RSA PSS" ) );
+            md_alg = MBEDTLS_MD_SHA256;
+            sig_alg = MBEDTLS_PK_RSASSA_PSS;
+            break;
+#endif /* MBEDTLS_X509_RSASSA_PSS_SUPPORT */
+        case MBEDTLS_TLS13_SIG_RSA_PKCS1_SHA256:
```

Comment:
This signature algorithm in TLS 1.3 is tolerated only for signatures in Certificate and should not be used for the signature in CertificateVerify (at least it is highly recommended). MBEDTLS_TLS13_SIG_RSA_PKCS1_SHA256 is advertised in the signature algorithms extension only to state that we support it for signatures in certificates.
Thus please remove this.